frame/pablo/src/lib.rs

```diff
@@ -299,40 +346,302 @@ pub mod pallet {
 		/// Emits `LiquidityRemoved` event when successful.
 		#[pallet::weight(10_000)]
 		pub fn remove_liquidity(
-			_origin: OriginFor<T>,
-			_pool_id: T::PoolId,
-			_lp_amount: T::Balance,
-			_min_base_amount: T::Balance,
-			_min_quote_amount: T::Balance,
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			lp_amount: T::Balance,
+			min_base_amount: T::Balance,
+			min_quote_amount: T::Balance,
 		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let _ = <Self as Amm>::remove_liquidity(
+				&who,
+				pool_id,
+				lp_amount,
+				min_base_amount,
+				min_quote_amount,
+			)?;
 			Ok(())
 		}
 	}
 
 	impl<T: Config> Pallet<T> {
-		pub(crate) fn do_create_pool(pool: PoolOf<T>) -> Result<T::PoolId, DispatchError> {
-			let pool_id =
-				PoolCount::<T>::try_mutate(|pool_count| -> Result<T::PoolId, DispatchError> {
-					let pool_id = *pool_count;
-					Pools::<T>::insert(pool_id, pool.clone());
-					*pool_count = pool_id.safe_add(&T::PoolId::one())?;
-					Ok(pool_id)
-				})?;
-			match pool {
-				PoolConfiguration::StableSwap(stable_swap_pool_info) => {
-					let pool = stable_swap_pool_info;
-					Self::deposit_event(Event::PoolCreated { pool_id, owner: pool.owner.clone() });
+		#[transactional]
+		pub(crate) fn do_create_pool(
+			who: &T::AccountId,
+			init_config: PoolInitOf<T>,
+		) -> Result<T::PoolId, DispatchError> {
+			match init_config {
+				PoolInitConfiguration::StableSwap {
+					pair,
+					amplification_coefficient,
+					fee,
+					protocol_fee,
+				} => {
+					let pool_id = StableSwap::<T>::do_create_pool(
+						&who,
+						pair,
+						amplification_coefficient,
+						fee,
+						protocol_fee,
+					)?;
+					Self::deposit_event(Event::PoolCreated { owner: who.clone(), pool_id });
+
 					Ok(pool_id)
 				},
 			}
 		}
 
-		pub(crate) fn get_pool(pool_id: T::PoolId) -> Result<PoolOf<T>, DispatchError> {
+		pub(crate) fn get_pool(
+			pool_id: T::PoolId,
+		) -> Result<PoolConfigurationOf<T>, DispatchError> {
 			Pools::<T>::get(pool_id).ok_or_else(|| Error::<T>::PoolNotFound.into())
 		}
 
 		pub(crate) fn account_id(pool_id: &T::PoolId) -> T::AccountId {
 			T::PalletId::get().into_sub_account(pool_id)
 		}
+
+		fn do_compute_swap(
```

Comment:
Seems this belongs in the `stableswap.rs`?

Reply:
it can be used across all DEXs, but can be specialized if all DEXs does not want to follow same maths for fees. Like in this case protocol_fee is % of lp_fee. (same as curve-amm). If we want something different in uniswap-v2 (like protocol_fee + lp_fee) is total fee then in future PR we can refactor this to have different specialized versions.